src/main/java/com/es/phoneshop/web/ProductDemoDataServletContextListener.java

```diff
@@ -0,0 +1,199 @@
+package com.es.phoneshop.web;
+
+import com.es.phoneshop.dao.CustomProductDao;
+import com.es.phoneshop.model.product.PriceChange;
+import com.es.phoneshop.model.product.Product;
+import jakarta.servlet.ServletContextEvent;
+import jakarta.servlet.ServletContextListener;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Currency;
+import java.util.Date;
+import java.util.List;
+
+public class ProductDemoDataServletContextListener implements ServletContextListener {
+
+    @Override
+    public void contextInitialized(ServletContextEvent servletContextEvent) {
+        boolean shouldInsertData = Boolean.parseBoolean(servletContextEvent.getServletContext().getInitParameter("insertDemoData"));
+        if (shouldInsertData) {
+            insertDataToProductDao();
+        }
+    }
+
+    private void insertDataToProductDao() {
+        CustomProductDao customProductDao = CustomProductDao.getInstance();
```

Comment:
you can store dao as a private listener field and initialize it outside the insertDataToProductDao(), method name says that we are only inserting data